pkg/reconciler/knativeeventing/controller.go

```diff
@@ -53,31 +54,31 @@ func NewController(
 ) *controller.Impl {
 	knativeEventingInformer := knativeEventinginformer.Get(ctx)
 	deploymentInformer := deploymentinformer.Get(ctx)
-
-	c := &Reconciler{
-		Base:                  rbase.NewBase(ctx, controllerAgentName, cmw),
-		knativeEventingLister: knativeEventingInformer.Lister(),
-		eventings:             sets.String{},
-	}
+	logger := logging.FromContext(ctx)
 
 	koDataDir := os.Getenv("KO_DATA_PATH")
 
 	cfg, err := sharedmain.GetConfig(*MasterURL, *Kubeconfig)
 	if err != nil {
-		c.Logger.Error(err, "Error building kubeconfig")
+		logger.Error(err, "Error building kubeconfig")
 	}
 
-	config, err := mfc.NewManifest(filepath.Join(koDataDir, "knative-eventing/"), cfg,
-		mf.UseLogger(zapr.NewLogger(c.Logger.Desugar()).WithName("manifestival")))
+	config, err := mfc.NewManifest(filepath.Join(koDataDir, "knative-eventing/"),
+		cfg,
+		mf.UseLogger(zapr.NewLogger(logger.Desugar()).WithName("manifestival")))
 	if err != nil {
-		c.Logger.Error(err, "Error creating the Manifest for knative-eventing")
+		logger.Error(err, "Error creating the Manifest for knative-serving")
 		os.Exit(1)
 	}
 
-	c.config = config
-	impl := controller.NewImpl(c, c.Logger, reconcilerName)
+	c := &Reconciler{
+		kubeClientSet: kubeclient.Get(ctx),
+		config:        config,
+		eventings:     sets.NewString(),
+	}
+	impl := knereconciler.NewImpl(ctx, c)
 
-	c.Logger.Info("Setting up event handlers for %s", reconcilerName)
+	logger.Info("Setting up event handlers")
```

Comment:
Do we need to the keep it the same as before by incuding reconcilerName? Or do we have other reasons to remove it?

Reply:
Good point, I shall remove the globals. I don't think there's any value in these anyway. The logger will be setup to have the right context straight away.

Reply:
Logs look like this:

```
{"level":"info","ts":"2020-04-27T15:49:54.622Z","logger":"eventing-operator","caller":"knativeeventing/controller.go:76","msg":"Setting up event handlers"}
```

There's enough context there imo.